### ISSUE TITLE
convert_filter only executed when node_filters are present

### ISSUE THREAD

**Issue by grekko** [gjtorikian/html-pipeline#391]
Hey hey, I'm playing around with `html-pipeline` and noticed that the following snippet does not run the `convert_filter`:

```rb
require "html_pipeline/convert_filter/markdown_filter"
require "selma"
markdown = "1. Foo\n2. Bar"
result = HTMLPipeline.new(convert_filter: HTMLPipeline::ConvertFilter::MarkdownFilter.new).call(markdown)
{
    :output => "1. Foo\n2. Bar"
}
```

vs. when defining *any* node_filter like so:

```rb
require "html_pipeline/convert_filter/markdown_filter"
require "html_pipeline/node_filter/https_filter"
require "selma"
markdown = "1. Foo\n2. Bar"
result = HTMLPipeline.new(
  convert_filter: HTMLPipeline::ConvertFilter::MarkdownFilter.new,
  node_filters: [HTMLPipeline::NodeFilter::HttpsFilter.new],
).call(markdown)
{
    :output => "<ol>\n<li>Foo</li>\n<li>Bar</li>\n</ol>"
}
```

AFAIU the `html`-result will not be used unless any `@node_filters` are present. See https://github.com/gjtorikian/html-pipeline/blob/74766e75a4c65a3d071faea84fc630d3fe217ab1/lib/html_pipeline.rb#L168-L179

**Comment by grekko** [comment]
Just let me know if you like to have a PR for that.

**Comment by franciscodiazydiaz** [comment]
Hey @grekko, I'm in the process of updating to v3 and just encountered the same issue. I was wondering if this was an intentional behavior. In my case, I only need the `convert_filter`.

**Comment by gjtorikian** [comment]
Completely unintentional—thank you for the report. Will be fixed monetarily in 3.0.2!
